### ISSUE TITLE
Addendum annotations not working in combination with eAccelerator (or other opcode cache)

### ISSUE THREAD

**Issue by PerWiklander** [bjorn-ali-goransson/PostTypeBuilder#4]
Addendum uses the built in docblock reading capabilities of PHP, but when an opcode cache is added to the mix there is no meta data about the classes to read any more.

Please see here:
http://php.net/manual/en/reflectionclass.getdoccomment.php#94061

In this case Addendum should go into manual parsing mode, and it does, but that does not work, at least not for my classes. Have you tested this with PostTypeBuilder? You can override the check here to simulate it:

https://github.com/bornemix/PostTypeBuilder/blob/master/addendum/annotations.php#L354

just return `true` instead.

This whole thing (that took me days to figure out after my project was broken after deploying to WP Engine) made me think that I would like an aproach where a build script reads the class files with annotations (just like it works now) and then generates PHP code for production use. All this magical annotation parsing at runtime feels a bit slow to me. It would probably make debugging easier as well if there is no magical stuff happening, the code is a bit hard to follow now at runtime. I had to investigate several other post type plugins to solve my project in a hurry and I found none that was as easy and well documented as PostTypeBuilder.

But that is a completely separate thing :-)

**Comment by bjorn-ali-goransson** [comment]
Hopefully I'll be able to look at it tonight.
Den 15 jan 2013 04:34 skrev "Per Wiklander" notifications@github.com:

**Comment by bjorn-ali-goransson** [comment]
The "raw mode" is a bit strange... it checks wether it can get its own doc comment (/*\* Raw mode test*/), and if it doesn't, well then we have a problem getting DocComments. So it switches to raw mode as you said. But even then it depends on ReflectionClass::getFileName, which needs to work.

So I'll carry on testing with raw mode I guess.

As to your problem, look what I found on the internet: "When using Reflection::getDocComment with eAccelerator it will always fail. To cache doc comments, eAccelerator has a configure switch `--with-eaccelerator-doc-comment-inclusion]."

**Comment by bjorn-ali-goransson** [comment]
Also, another solution (which needs to work in production scenarios anyway) could be to save the meta-data in plain .php files for inclusion. And then try to invalidate them if the php files are touched or saved.

**Comment by PerWiklander** [comment]
I'm deploying to WP Engine and I am pretty sure they do not make any changes to their config just because I ask them, let alone any recompiling of PHP extensions.

**Comment by bjorn-ali-goransson** [comment]
Right. This has to do with raw mode not handling namespaces consistently;
it should be fixed tonight, no worries.
Den 16 jan 2013 06:53 skrev "Per Wiklander" notifications@github.com:

**Comment by PerWiklander** [comment]
Ok, good with a solution. What do you think about caching the meta data in PHP-files for prod-mode though? As a new feature that is.

**Comment by bjorn-ali-goransson** [comment]
Yeah, absolutely. I've planned for it since the start...
Den 16 jan 2013 08:32 skrev "Per Wiklander" notifications@github.com:

**Comment by bjorn-ali-goransson** [comment]
Ok, namespaces are in with afd64b834635608c4258ea258116fff4d9b17845
